src/main/java/de/komoot/photon/elasticsearch/Searcher.java

```diff
@@ -57,12 +71,32 @@ public Searcher(Client client) {
 		if(lon != null) params.put("lon", lon);
 		if(lat != null) params.put("lat", lat);
 
-		StrSubstitutor sub = new StrSubstitutor(params.build(), "${", "}");
-		if(lon != null && lat != null) {
-			query = sub.replace(queryLocationBiasTemplate);
-		} else {
-			query = sub.replace(queryTemplate);
-		}
+        this.setupSearch(lon,lat,tagKey,tagValue);
```

Comment:
Will you consider this as a future improvement to be taken up as a separate issue? 

I did feel the same way as you do about the battery of if statements. Which is why I asked my question in a previous [comment](https://github.com/komoot/photon/issues/68#issuecomment-71639283). So, then, having just one JSON is ok?

Further, if you will consider my suggestion: currently, photon code is very tightly coupled and difficult to refactor without adding more if-then-elses. For example, in [App.java](https://github.com/komoot/photon/blob/30f626d92f7ff658a17c9372a8af1432dbe62e45/src/main/java/de/komoot/photon/App.java#L135), The Searcher, RequestHandler are concrete classes tied together in code. Within Searcher too, the search method does parameter replacement, query execution and then also delegates to conversion and cleansing. If these are refactored into different classes, photon code could become less bug-prone and we may be able to avoid these if-then-elses.

Of course, I don't mean to review your code, it is a very useful project as it is. I just wanted to offer my thoughts on the same lines of making the code/json less error prone.

Let me know if you would consider doing all this in a different issue as an improvement. 

Thanks!